torch/optim/adam.py

```diff
@@ -377,7 +382,11 @@ def _single_tensor_adam(params: List[Tensor],
         step_t += 1
 
         if weight_decay != 0:
-            grad = grad.add(param, alpha=weight_decay)
+            if not decoupled_weight_decay:
```

Comment:
nit: Would prefer flipping the if and else here so it would be like `if decoupled_weight_decay:` for slightly more clarity.